staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -129,9 +130,31 @@ func ValidateNoNewFinalizers(newFinalizers []string, oldFinalizers []string, fld
 
 // ValidateImmutableField validates the new value and the old value are deeply equal.
 func ValidateImmutableField(newVal, oldVal interface{}, fldPath *field.Path) field.ErrorList {
+	return ValidateImmutableFieldWithErr(newVal, oldVal, field.Invalid(fldPath, newVal, FieldImmutableErrorMsg))
+}
+
+// ValidateImmutableFieldWithErr validates the new value and the old value are deeply equal.
+// It accepts an err *field.Error parameter as base error. If the result is not equal, ValidateImmutableFieldWithErr will append additional error information
+// in err.detail and return this error.
+func ValidateImmutableFieldWithErr(newVal, oldVal interface{}, err *field.Error) field.ErrorList {
```

Comment:
One thing that you might want to fix later (not in this PR), is that it makes no sense to have `ValidateIMmutableField` and `ValidateImmutableFieldWithErr` to return a `field.ErrorList` when they clearly can only generate one error.